AI-256: Add sync action execution tool (#413)
keboola/mcp-server: src/keboola_mcp_server/resources/prompts/project_system_prompt.md

  NEVER like this: `params = ci.configuration.parameters.get("user_properties", {})`

### Creating or Updating Component Configurations

> #### Discovery & Schema

Reviewer: I think that the markdown format mandates an empty line before a heading. Please add those empty lines. They also help human readability.

#### Creating a Configuration
1. Build the configuration based on the schema and/or examples
2. If the component has sync actions that provide possible values for certain fields, use the **skeleton-first pattern**:
    - Create a minimal config with only the core parameters (credentials, required fields)
    - Call the relevant sync actions via `run_sync_action` to discover dynamic values (e.g., available tables, schemas, columns)
    - Update the config with the resolved values
3. If no such sync actions exist, create the full configuration directly
#### Updating a Configuration
Always fetch the existing config first. Apply only targeted changes and preserve all unmodified fields.
#### Sync Actions
Sync actions execute the component synchronously to perform specific tasks such as testing connections, listing remote tables or columns, and validating credentials. They are typically used to dynamically load available values or perform validation.

###### Finding Available Actions
Available sync actions for a component are listed in the `sync_actions` field returned by `get_components`.

###### Scoping: Root vs. Row Level

> Sync actions can exist at the root config level or at the configuration row level:

Reviewer: ```suggestion
Sync actions can operate on the component's root or row configuration:
```

> - **Root-level actions** (e.g., `testConnection`) apply to the entire configuration — pass the config ID only

Reviewer: Fixed in b6cf200a
